telemetry/definitions/commonDefinitions.json

```diff
@@ -47,6 +61,11 @@
             "type": "string",
             "description": "A generic name metadata"
         },
+        {
+            "name": "filenameExt",
+            "type": "string",
+            "description": "Filename extension (examples: .txt, .yml, .yaml, .asl.yaml, ...), or empty string if the filename does not contain dot (.) between two chars"
```

Comment:
Do we also want a generic "file opened" metric to see what kinds of files people are editing? This would only track file extension, for privacy (and we could set a hard limit of 8 chars).

Reply:
I don't think the Toolkits should emit something so open ended. As a user this would feel too invasive, because it doesn't appear to have relevance to the AWS extension. IMO it would be preferable to measure something specific that we intend to take action on.

Reply:
consolas is doing this so let's model it

Reply:
> As a user this would feel too invasive, because it doesn't appear to have relevance to the AWS extension

It is only the extension, and only for filenames we support.